### ISSUE TITLE
OwnershipCard component links don't filter on fully qualified entity refs

### ISSUE THREAD

**Issue by jpeach** [backstage/backstage#11347]
## Expected Behavior

Clicking on a component type box in the OwnershipCard should take you to a list of all the components of the given type with a matching owner.

## Actual Behavior

If the owner is in a namespace (e.g. a ref like `group:teams/foo`), the namespace is not included in the filter, so no results are returned. Manually adding the namespace to the filter makes it work.

## Steps to Reproduce

1. Create a group in a namespace
2. Create a component owned by that group
3. Navigate to the group entity page and click on an item in the Ownership panel.

## Context

I used namespaces to import groups because otherwise we would have name collisions.

## Your Environment

- Output of `yarn backstage-cli info`: 
```text
OS:   Darwin 21.4.0 - darwin/arm64
node: v16.15.0
yarn: 1.22.18
cli:  0.17.0 (installed)

Dependencies:
  @backstage/app-defaults                            1.0.1
  @backstage/backend-common                          0.13.2
  @backstage/backend-tasks                           0.3.0
  @backstage/catalog-client                          1.0.1
  @backstage/catalog-model                           1.0.1
  @backstage/cli-common                              0.1.8
  @backstage/cli                                     0.17.0
  @backstage/config-loader                           1.1.0
  @backstage/config                                  1.0.0
  @backstage/core-app-api                            1.0.1
  @backstage/core-components                         0.9.3
  @backstage/core-plugin-api                         1.0.1
  @backstage/dev-utils                               1.0.1
  @backstage/errors                                  1.0.0
  @backstage/integration-react                       1.0.1
  @backstage/integration                             1.1.0
  @backstage/plugin-api-docs                         0.8.4
  @backstage/plugin-app-backend                      0.3.31
  @backstage/plugin-auth-backend                     0.13.0
  @backstage/plugin-auth-node                        0.2.0
  @backstage/plugin-catalog-backend-module-bitbucket 0.1.2
  @backstage/plugin-catalog-backend                  1.1.1
  @backstage/plugin-catalog-common                   1.0.1
  @backstage/plugin-catalog-graph                    0.2.16
  @backstage/plugin-catalog-import                   0.8.7
  @backstage/plugin-catalog-react                    1.0.1
  @backstage/plugin-catalog                          1.1.0
  @backstage/plugin-kubernetes-common                0.2.9
  @backstage/plugin-kubernetes                       0.6.4
  @backstage/plugin-org                              0.5.4
  @backstage/plugin-pagerduty                        0.3.31
  @backstage/plugin-permission-common                0.6.0
  @backstage/plugin-permission-node                  0.6.0
  @backstage/plugin-permission-react                 0.4.0
  @backstage/plugin-proxy-backend                    0.2.25
  @backstage/plugin-scaffolder-backend               1.1.0
  @backstage/plugin-scaffolder-common                1.0.1
  @backstage/plugin-scaffolder                       1.1.0
  @backstage/plugin-search-backend-node              0.6.0
  @backstage/plugin-search-backend                   0.5.1
  @backstage/plugin-search-common                    0.3.3
  @backstage/plugin-search-react                     0.1.0
  @backstage/plugin-search                           0.8.0
  @backstage/plugin-techdocs-backend                 1.1.0
  @backstage/plugin-techdocs-node                    1.1.0
  @backstage/plugin-techdocs-react                   0.1.0
  @backstage/plugin-techdocs                         1.1.0
  @backstage/plugin-user-settings                    0.4.3
  @backstage/release-manifests                       0.0.2
  @backstage/search-common                           0.3.3
  @backstage/test-utils                              1.0.1
  @backstage/theme                                   0.2.15
  @backstage/types                                   1.0.0
  @backstage/version-bridge                          1.0.1
```

**Comment by jpeach** [comment]
cc @renlord

**Comment by anicke** [comment]
I belive this was fixed in #11139.

**Comment by jpeach** [comment]
> I believe this was fixed in #11139.

Yep, I agree. Thanks @kuangp and @anicke!
